### ISSUE TITLE
Generate request spec by default when generating scaffolds

### ISSUE THREAD

**Issue by klyonrad** [rspec/rspec-rails#2230]
Similar problem like in #2056. Upon using `rails generate scaffold` it would be good if controller tests would not get generated and instead request specs are generated.

**Comment by JonRowe** [comment]
Now that #2222 is merged do you fancy a go at this as well @klyonrad ?

**Comment by klyonrad** [comment]
Yes, that's why I created this issue 😉

**Comment by klyonrad** [comment]
Hey @JonRowe do you happen to have an idea why the Scaffold Generator creates the empty request spec?

I think my previous PR had this effect

See

```

rails generate scaffold product title:string body:text published:boolean -p

....
      invoke    rspec
      create      spec/controllers/products_controller_spec.rb
      create      spec/views/products/edit.html.erb_spec.rb
      create      spec/views/products/index.html.erb_spec.rb
      create      spec/views/products/new.html.erb_spec.rb
      create      spec/views/products/show.html.erb_spec.rb
      create      spec/routing/products_routing_spec.rb
      invoke      rspec
      create        spec/requests/products_spec.rb

...
```

I suspect that it caused because the scaffolds generator executes the controller Generator and the Controller Generator has the default option for the request spec from my previous PR

**Comment by JonRowe** [comment]
I am unsure, I can see it happens although it is not empty for me.., the fact there is extra indented output seems wrong too

**Comment by JonRowe** [comment]
So I investigated, and I'm pretty certain this happens even without your changes...

**Comment by JonRowe** [comment]
In fact its `hook_for` that triggers it

**Comment by klyonrad** [comment]
> So I investigated, and I'm pretty certain this happens even without your changes...

@JonRowe Are you sure or did I maybe make things confusing? I didn't mean the PR that is linked to this PR, but rather #2222 is the culprit. Should I open a separate issue to avoid confusion?

My theory is that the added `class_option :request_specs,    type: :boolean, default: true, desc: "Generate request specs"` **trickles over while the scaffold generator executes controller generator.** This theory comes from the fact that scaffold generator acts weird even when I use the current stable version (3.8.3) and add the `--request_specs` option through the command line usage, for example `rails generate scaffold product title:string body:text published:boolean -p --request_specs`.

> In fact its `hook_for` that triggers it

It was not abvious to me why the `hook_for` was added there. I went into the git history in the file and landed at the historic initial commits where josé valim (❤️ ) is mentioned f06aa507280f8f9e6316c9bd5f72e640269a49b1

**Comment by JonRowe** [comment]
> @JonRowe Are you sure or did I maybe make things confusing? I didn't mean the PR that is linked to this PR, but rather #2222 is the culprit. Should I open a separate issue to avoid confusion?

I'm sure, I reverted #2222, and it still happened. What happens is `hook_for` is a trigger for activating other generators, and it runs the equivalent of `rails generate integration product` which is currently implemented as a request spec.

**Comment by JonRowe** [comment]
Fixed in #2288
